netebpfext/net_ebpf_ext_sock_addr.c

```diff
@@ -1372,22 +1372,27 @@ _ebpf_sock_addr_context_create(
     size_t context_size_in,
     _Outptr_result_maybenull_ void** context)
 {
+    NET_EBPF_EXT_LOG_ENTRY();
+
     ebpf_result_t result;
     bpf_sock_addr_t* sock_addr_ctx = NULL;
     if (!context) {
         result = EBPF_INVALID_ARGUMENT;
         goto Done;
     }
-    *context = NULL;
```

Comment:
Why was this removed?  Seems that it would be good to NULL on failure (so even better would be to move it up to 1376.

Also line 1373 annotation says it can be NULL on success (the `_result_maybenull_`) but I don't see how that can happen so maybe the annotation should just be `_Outptr_`?

Reply:
Originally this was there because the code analysis was warning about uninitialized out parameters as it didn't detect the success condition. Removed it when I switched to ebpf_result_t (which has ```_Return_type_success_(return == EBPF_SUCCESS)```).

But I agree that it's worth having to make sure we always set output on success.